Packages/Sequence-Unity/Sequence/SequenceFrontend/Scripts/UI/Demos/MarketplaceItemDetailsPage.cs

```diff
@@ -0,0 +1,57 @@
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using Sequence.Marketplace;
+
+namespace Sequence.Demo
+{
+    public class MarketplaceItemDetailsPage : UIPage
+    {
+        [SerializeField] UIPanel _getCheckoutOptionsPanel;
+        [SerializeField] Image _image;
+        [SerializeField] private TextMeshProUGUI _nameText, _priceText, _descriptionText, _attributesText;
+        CollectibleOrder _order;
+
+        protected override void Awake()
+        {
+            base.Awake();
+        }
+        public override void Open(params object[] args)
+        {
+            ClearPage();
+
+            base.Open(args);
+
+            foreach (var item in args)
+            {
+                if (item is CollectibleOrder order) _order = order;
+                break;
+            }
+            FillPage();
+        }
+
+        void ClearPage()
+        {
+            _image.sprite = null;
+            _nameText.text = string.Empty;
+            _priceText.text = string.Empty;
+            _descriptionText.text = string.Empty;
+            _attributesText.text = string.Empty;
+            _order = null;
+        }
+
+        async void FillPage()
+        {
+            var texture = await UnityWebRequestExtensions.DownloadImage(_order.metadata.image);
```

Comment:
This approach is totally fine. For future, you can also use the SpriteFetcher static class